include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -449,7 +449,10 @@ KOKKOS_INLINE_FUNCTION ddc::DiscreteElement<NonUniformBSplines<Tag, D>> NonUnifo
                 d += a(k, s2) * ndu(pk, r);
             }
             derivs(r, k) = d;
-            std::swap(s1, s2);
+            // swap s1 <-> s2;
+            auto tmp = s1;
+            s1 = s2;
+            s2 = tmp;
```

Comment:
It looks like a manual inline of `std::swap`, why do you need that ?

Reply:
There is a warning because std::swap is host function called from KOKKOS_FUNCTION

Reply:
I see, indeed `std::swap` is not constexpr until C++20